### PR TITLE
chore: refactor postgresql addon and add scripts ut 

### PR DESCRIPTION
**Pull request by Y-Rookie**
1. Refactor lifecycle actions based on the new API and kbagent.
2. Remove all references to built-in KB_XX environment variables.
3. Eliminate dependencies on the Downward API and other similar dependencies.
4. Refactor scripts and add corresponding unit tests (ut).

### CODE REVIEW

File: addons/postgresql/values.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ fullnameOverride: ""
 image:
   # refer: https://github.com/apecloud-inc/spilo/blob/main/postgres-appliance/Dockerfile
   # this is the image of postgresql, which is built from Dockerfile of apecloud-inc/spilo forked from zalando/spilo.
-  registry: docker.io
+  registry: apecloud-registry.cn-zhangjiakou.cr.aliyuncs.com
```
Comment (ldming) [inline addons/postgresql/values.yaml:17]: 这里应该是 @JashBook 特意改成了 docker io，默认使用 docker hub.

File: addons/postgresql/values.yaml [hunk 2]
```diff
@@ -104,7 +105,7 @@ pgbouncer:
     # refer: https://hub.docker.com/r/bitnami/pgbouncer
     # if the value of pgbouncer.image.registry is not specified using `--set`, it will be set to the value of 'image.registry' by default
     registry: ""
-    repository: bitnami/pgbouncer
+    repository: apecloud/pgbouncer
```
Comment (JashBook) [inline addons/postgresql/values.yaml:108]: Same as above

File: addons/postgresql/values.yaml (hunk 1 above)
Comment (Y-Rookie) [reply addons/postgresql/values.yaml]: > @JashBook changed this registry to docker hub, default use docker.io

fixed, forget to revert, just for test

File: addons-cluster/postgresql/templates/_helpers.tpl [hunk 3]
```diff
@@ -18,7 +18,9 @@ Define postgresql ComponentSpec with ComponentDefinition.
   clusterDefinitionRef: postgresql
   topology: {{ .Values.mode }}
   componentSpecs:
-    - name: postgresql
+    - name: {{ include "postgresql-cluster.component-name" . }}
+      labels:
+        {{- include "postgresql-cluster.patroni-scope-label" . | indent 8 }}
```
Comment (xuriwuyun) [inline addons-cluster/postgresql/templates/_helpers.tpl:23]: how to specify the pg version when install the chart?

Comment (Y-Rookie) [reply addons-cluster/postgresql/templates/_helpers.tpl:23]: > how to specify the pg version when install the chart?

This is not within the scope of this PR. The version will be specified through serviceVersion in the future, but the management of serviceVersion requires further design.

**Review by ldming — APPROVED** [review]

**Review by xuriwuyun — APPROVED** [review]

**Review by ldming — APPROVED** [review]

**Pull request merged**